src/ol/interaction/Select.js

```diff
@@ -208,7 +213,18 @@ class Select extends Interaction {
      */
     this.hitTolerance_ = options.hitTolerance ? options.hitTolerance : 0;
 
-    const featureOverlay = new VectorLayer({
+    /**
+     * @private
+     * @type {boolean}
+     */
+    this.userSpecifiedFeatureOverlay_ = typeof options.featureOverlay !== 'undefined' &&
```

Comment:
I'd prefer a simpler name, e.g. `this.userLayer_`. Also, please compute the boolean value with
```js
this.userLayer_ = options.featureOverlay !== undefined;
```
instead.